src/Middleware/HoneybadgerContext.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace Honeybadger\HoneybadgerLaravel\Middleware;
+
+use Closure;
+use Honeybadger\Contracts\Reporter;
+use Illuminate\Support\Facades\Route;
+
+class HoneybadgerContext
+{
+    /**
+     * @var \Honeybadger\Honeybadger;
+     */
+    protected $honeybadger;
+
+    /**
+     * @param  \Honeybadger\Contracts\Reporter  $honeybadger
+     */
+    public function __construct(Reporter $honeybadger)
+    {
+        $this->honeybadger = $honeybadger;
+    }
+
+    /**
+     * Handle an incoming request.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  \Closure  $next
+     * @return mixed
+     */
+    public function handle($request, Closure $next)
+    {
+        if (app()->bound('honeybadger')) {
+            $this->setUserContext($request);
+            $this->setRouteActionContext();
+        }
+
+        return $next($request);
+    }
+
+    private function setRouteActionContext()
+    {
+        if (Route::getCurrentRoute()) {
+            $routeAction = explode('@', Route::getCurrentRoute()->getActionName());
+
+            $this->honeybadger->setComponent($routeAction[0] ?? null);
+            $this->honeybadger->setAction($routeAction[1] ?? null);
```

Comment:
On multiple sites, running Laravel 5.8.35, after upgrading to `1.7.0` getting the following error on invalid URLs (404 pages):
```
Argument 1 passed to Honeybadger\Honeybadger::setAction() must be of the type string, null given, called in /vendor/honeybadger-io/honeybadger-laravel/src/Middleware/HoneybadgerContext.php on line 47
```

Reply:
This issue has been reported, will have a fix today